promql/promqltest/testdata/type_and_unit.test

```diff
@@ -217,15 +217,15 @@ eval instant at 50m rate(http_requests_total[25m]) * 25 * 60
 
 eval instant at 50m http_requests_total{group="canary"} and http_requests_total{instance="0"}
 	http_requests_total{__type__="gauge", __unit__="not-request", group="canary", instance="0", job="api-server"} 300
-	http_requests_total{__type__="counter", __unit__="", group="canary", instance="0", job="app-server"} 700
+	http_requests_total{__type__="counter", group="canary", instance="0", job="app-server"} 700
```

Comment:
@bwplotka could you look at this change please?  Is it essential that `__unit__=""` work inside Prometheus?  Blank values are not usually allowed.

Reply:
Empty labels are not disallowed per se, but they are considered equivalent to unset labels, so we have always "normalized" them away whenever we store them, display them, etc. See also https://prometheus.io/docs/concepts/data_model/:

"Labels with an empty label value are considered equivalent to labels that do not exist."

Reply:
IMO, being able to do `__unit__=""` is a nice-to-have, since that would allow differentiating between metrics without a unit, and those with one if there is a naming collision.  But it is probably not a very common case. More often I would expect to see this used when base and non-base units are mixed (e.g. seconds and milliseconds).

Maybe we could treat `foo=""` as "foo is the empty string _or is unset_" to be consistent with empty string and unset being equivalent? But otherwise I think it is OK to treat this the same way we treat other labels.

It seems like too disruptive a change to move from words to UCUM for units to be worth it...

Reply:
Please note this is a unit test checking that a label with an actual blank value appears in the result of a PromQL expression.
I think this was an error, should not be allowed, and should not be in the test.

As far as I can see none of the other comments in this thread are relevant to the question I wanted to ask.